### PR TITLE
Migrate ATT Support

### PR DESCRIPTION
**Pull request by Bawnorton**
Don't merge until I publish ATT's latest release as this change will break ATT's current compat mixin

### CODE REVIEW

**Comment by Bawnorton** [comment]
ATT 4.0.1 is public, can merge

**Comment by AmyMialee** [comment]
How would one build a jar of this? This is a wild gradle setup.

**Comment by Bawnorton** [comment]
https://stonecutter.kikugie.dev/

File: src/main/kotlin/dev/kikugie/elytratrims/common/compat/AllTheTrimsCompat.kt [hunk 1]
```diff
@@ -33,11 +41,68 @@ object AllTheTrimsCompat {
...
+                "trims/models/elytra/%s_%s".format(
+                    trim.pattern.value().assetId().path,
+                    AllTheTrims.DYNAMIC
+                )
+            )
+        } else {
+            renderer.getModelId(sprite)
+        }
+        val renderLayer = if (renderer.useLegacyRenderer(sprite)) {
+            Compat.getElytraTrimsCompat().orElseThrow().elytraTrimRenderLayer
```
Comment (kikugie) [inline src/main/kotlin/dev/kikugie/elytratrims/common/compat/AllTheTrimsCompat.kt]: This reference doesn't exist in 4.0.1 anymore. What should the replacement be?

Comment (Bawnorton) [reply src/main/kotlin/dev/kikugie/elytratrims/common/compat/AllTheTrimsCompat.kt]: That's just `ElytraTrimsAPI#getElytraLayer`, I'll push

**Pull request merged**